pilot/pkg/model/mesh.go

```diff
@@ -0,0 +1,785 @@
+
+	// Key Istio attribute names for mapping endpoints to subsets.
+	// For details, please see
+	// https://istio.io/docs/reference/config/mixer/attribute-vocabulary.html
+
+	// UID is the attribute name for the mesh unique, platform-specific
+	// unique identifier for the server instance of the destination service. Mesh
+	// uses the label to uqniquely identify the endpoint when it receives updates
+	// from the Controller. Example: kubernetes://my-svc-234443-5sffe.my-namespace
+	// No two instances running anywhere in the mesh can have the same value for
+	// UID.
+	UID destinationAttribute = "destination.uid"
+
+	// SERVICE represents the fully qualified __CANONOCAL__ name of the Istio
+	// service, ex: "my-svc.my-namespace.svc.cluster.local".  Mesh uses
+	// this label to identify endpoints belonging to the same service for XDS queries
+	// involving service names. The attribute value is expected to be mesh unique,
+	// meaning services having this name are semantically understood to be the same
+	// modulo version and other labels, irrespective of the platform under which
+	// they are run, the clusters or availability zones they run in.
+	SERVICE destinationAttribute = "destination.service"
+
+	// NAME represents the short part of the service name, ex: "my-svc".
+	NAME destinationAttribute = "destination.name"
+
+	// NAMESPACE represents the namespace part of the destination service,
+	// example: "my-namespace".
+	NAMESPACE destinationAttribute = "destination.namespace"
+
+	// IP represents the IP address of the server instance, example 10.0.0.104.
+	// This IP is expected to be reachable from __THIS__ pilot. No distinction
+	// is being made for directly reachable service instances versus those
+	// behind a VIP. Istio's health discovery service will ensure that this
+	// endpoint's capacity is correctly reported accordingly.
+	IP destinationAttribute = "destination.ip"
+
+	// PORT represents the recipient port on the server IP address, Example: 443
+	PORT destinationAttribute = "destination.port"
+
+	// DOMAIN represents the domain portion of the service name, excluding
+	// the name and namespace, example: svc.cluster.local
+	DOMAIN destinationAttribute = "destination.domain"
+
+	// USER represents the __CANONICAL__ Mesh user running the destination
+	// application, example: service-account-foo. The user's identity is
+	// assumed to be trusted within the entire mesh, irrespective of which
+	// platform, cluster or availability zone this instance runs in.
+	USER destinationAttribute = "destination.user"
+)
+
+var (
+	destinationAttrSet = map[string]bool{
+		UID.stringValue():       true,
+		SERVICE.stringValue():   true,
+		NAME.stringValue():      true,
+		NAMESPACE.stringValue(): true,
+		IP.stringValue():        true,
+		PORT.stringValue():      true,
+		DOMAIN.stringValue():    true,
+		USER.stringValue():      true,
+	}
+
+	canonicalIstioDomain = "svc.cluster.local"
+	svcRegexp            = regexp.MustCompile(
```

Comment:
Well, we obviously must accept service names not using .svc.cluster.local ( bring your own name, other cluster names, etc). And I'm not sure the regex is exactly a DNS label - but maybe check if go standard library doesn't have one already. I think it's supposed to be 63 chars, etc.

Reply:
@costinm - yes that's via aliases. Talking to @ZackButcher we felt aliases satisfies what you need done. Keep in mind ... the canonical pieces of this is mostly for uniqueness within mesh. For on-prem to work correctly. A service running Consul should be identifiable in any control plane running Istio irrespective of whether this is Eureka or K8s running in Amazon etc...